helios-tools/src/main/java/com/spotify/helios/auth/ClientAuthenticationPluginLoader.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2014 Spotify AB.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.helios.auth;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterators;
+import com.google.common.collect.Lists;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.ServiceLoader;
+
+public class ClientAuthenticationPluginLoader {
+
+  public static List<ClientAuthenticationPlugin> loadAll() {
+    return loadAll(null);
+  }
+
+  public static List<ClientAuthenticationPlugin> loadAll(final Path pluginPath) {
+    final ServiceLoader<ClientAuthenticationPlugin> loader =
+        serviceLoaderForPath(pluginPath);
+    final List<ClientAuthenticationPlugin> plugins = Lists.newArrayList();
+    Iterators.addAll(plugins, loader.iterator());
+    return ImmutableList.copyOf(plugins);
```

Comment:
I think these 3 lines could just be `return ImmutableList.copyOf(loader.iterator())`